src/SFA.DAS.Forecasting.Jobs.Infrastructure/Logging/NLogConfiguration.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+using NLog;
+using NLog.Common;
+using NLog.Config;
+using NLog.Targets;
+using SFA.DAS.NLog.Targets.Redis.DotNetCore;
+
+namespace SFA.DAS.Forecasting.Jobs.Infrastructure.Logging
+{
+    public class NLogConfiguration
+    {
+        public void ConfigureNLog(IConfiguration configuration)
+        {
+            var appName = configuration.GetConnectionStringOrSetting("AppName");
+            var env = configuration.GetConnectionStringOrSetting("EnvironmentName");
+            var config = new LoggingConfiguration();
+
+            if (string.IsNullOrEmpty(env) || env.Equals("LOCAL", StringComparison.CurrentCultureIgnoreCase))
+            {
+                AddLocalTarget(config, appName);
+            }
+            else
+            {
+                AddRedisTarget(config, appName);
+            }
+
+            LogManager.Configuration = config;
+        }
+
+        private static void AddLocalTarget(LoggingConfiguration config, string appName)
+        {
+            InternalLogger.LogFile = Path.Combine(Directory.GetCurrentDirectory(), $"{appName}\\nlog-internal.{appName}.log");
+            var fileTarget = new FileTarget("Disk")
+            {
+                FileName = Path.Combine(Directory.GetCurrentDirectory(), $"{appName}\\{appName}.${{shortdate}}.log"),
+                Layout = "${longdate} [${uppercase:${level}}] [${logger}] - ${message} ${onexception:${exception:format=tostring}}"
+            };
+            config.AddTarget(fileTarget);
+
+            config.AddRule(GetMinLogLevel(), LogLevel.Fatal, "Disk");
+        }
+
+        private static void AddRedisTarget(LoggingConfiguration config, string appName)
+        {
+            var target = new RedisTarget
+            {
+                Name = "RedisLog",
+                AppName = appName,
+                EnvironmentKeyName = "EnvironmentName",
+                ConnectionStringName = "LoggingRedisConnectionString",
```

Comment:
Should `EnvironmentKeyName` and `ConnectionStringName` be set to string literals? Is there some sort of later substitution happening?

Reply:
Yes it picks these from configuration

Reply:
Looks like the name of environment variables 😄